### PR TITLE
Fix #5212: removing bug causing progress dialog to load forever

### LINKED ISSUE #5212: Edit depictions: "Please wait" forever
Steps to reproduce:
1. Tap on any already uploaded picture.
2. Tap the pencil icon of the "Depictions" section.
3. Add any depiction.
4. Tap "Save"
5. Popup saying "Please wait" appears and never disappears.

https://github.com/commons-app/apps-android-commons/assets/99590/c42fc94c-c3d1-41e4-aee1-e02a9956f0c6

Screenshot_20230417-222342

Note: The edit is actually saved and pressing Back lets you use the app without killing it.

### PR DESCRIPTION
**Pull request by TaiHaDev**
Fixes: #5212

Description:
The bug originates from the [updateDepictions](https://github.com/commons-app/apps-android-commons/blob/main/app/src/main/java/fr/free/nrw/commons/upload/depicts/DepictsPresenter.kt#L196) function, which manages depiction updates and observes the result with both success and error handlers. While it does initiate the depiction update in the databases, the subsequent call to the [showNotification](https://github.com/commons-app/apps-android-commons/blob/main/app/src/main/java/fr/free/nrw/commons/notification/NotificationHelper.java#L66) function leads to an error. This issue arises because API levels 31 and above require a more specific flag for pending intents.

Changes:
Introduced an if statement to assess the current API version. If the API version exceeds 31, the flag is set to immutable.

Testing:
Verified on API levels 28 and 33 without any issues.

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java [hunk 1]
```diff
@@ -505,6 +505,7 @@ private void onMediaRefreshed(Media media) {
                 .subscribe(this::onDepictionsLoaded, Timber::e)
         );
         // compositeDisposable.add(disposable);
+
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java:508]: Any particular reason for this change? If not, would you mind removing the new line?

File: app/src/main/java/fr/free/nrw/commons/upload/depicts/DepictsPresenter.kt [hunk 2]
```diff
@@ -224,9 +224,9 @@ class DepictsPresenter @Inject constructor(
                             Timber.e(
                                 "Failed to update depictions"
                             )
+
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/upload/depicts/DepictsPresenter.kt:227]: same

**Review by nicolas-raoul — CHANGES_REQUESTED** [review]

File: app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java (hunk 1 above)
Comment (TaiHaDev) [reply app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java]: I just added it by accident. I have removed it. Thank you!

File: app/src/main/java/fr/free/nrw/commons/upload/depicts/DepictsPresenter.kt (hunk 2 above)
Comment (TaiHaDev) [reply app/src/main/java/fr/free/nrw/commons/upload/depicts/DepictsPresenter.kt]: removed

**Review by nicolas-raoul — APPROVED** [review]
Tested on several pictures, it works great!
https://commons.wikimedia.org/w/index.php?title=File:Ikuchijima_211.jpg&diff=prev&oldid=812977952

Source code looks good too.

**Pull request merged**